src/projections.jl

```diff
@@ -253,6 +253,82 @@ function projection_on_set(d::DefaultDistance, v::AbstractVector{T}, ::MOI.DualE
     return v + projection_on_set(d, -v, MOI.ExponentialCone())
 end
 
+"""
+    projection_on_set(::DefaultDistance, v::AbstractVector{T}, ::MOI.PowerCone) where {T}
+
+projection of vector `v` on the power cone
+i.e. `K = {(x,y,z) | x^a * y^(1-a) >= |z|, x>=0, y>=0}`.
+
+References:
+* [Differential properties of Euclidean projection onto power cone]
+(https://link.springer.com/article/10.1007/s00186-015-0514-0), Prop 2.2
+"""
+function projection_on_set(::DefaultDistance, v::AbstractVector{T}, s::MOI.PowerCone) where {T}
+    _check_dimension(v, s)
+
+    if _in_pow_cone(v, s)
+        return v
+    end
+    if _in_pow_cone(-v, MOI.dual_set(s))
+        # if in polar cone Ko = -K*
+        return zeros(T, 3)
+    end
+    if abs(v[3]) <= 1e-10
+        return [max(v[1],0); max(v[2],0); 0.0]
+    end
+
+    return _solve_system_pow_cone(v, s; arg=false)
+end
+
+function _in_pow_cone(v::AbstractVector{T}, cone::MOI.PowerCone; tol=1e-10) where {T}
+    α = cone.exponent
+    return v[1] >= 0 && v[2] >= 0 && tol + v[1]^α * v[2]^(1-α) >= abs(v[3])
+end
+
+function _in_pow_cone(v::AbstractVector{T}, cone::MOI.DualPowerCone; tol=1e-10) where {T}
+    α = cone.exponent
+    return (
+        v[1] >= 0 && v[2] >=0 && tol + (v[1])^α * (v[2])^(1-α) >= α^α * (1-α)^(1-α) * abs(v[3])
+    )
+end
+
+function _solve_system_pow_cone(v::AbstractVector{T}, s::MOI.PowerCone; arg=true) where {T}
```

Comment:
This function could get a docstring explaining what is done here. Also, what does the arg keyword correspond to, is it dual / polar?

Reply:
Indication of whether to return the argument that solves the system or the associated projection, which requires a transformation. I wrote it this way to avoid duplicating `Phi_prod`, but on second thought it would be cleaner if this function always returns the solution to the system.

Reply:
yes you can return the two as a tuple (argument, solution)